mesa/space.py

```diff
@@ -118,29 +122,26 @@ def __init__(self, width: int, height: int, torus: bool) -> None:
         self.empties = set(itertools.product(*(range(self.width), range(self.height))))
 
     @staticmethod
-    def default_val() -> None:
-        """ Default value for new cell elements. """
-        return None
+    def default_val() -> Content:
+        pass
 
-    def __getitem__(self, index: int) -> List[GridContent]:
+    def __getitem__(self, index: int) -> List[Content]:
         return self.grid[index]
 
-    def __iter__(self) -> Iterator[GridContent]:
+    def __iter__(self) -> Iterator[Content]:
         """
         create an iterator that chains the
         rows of grid together as if one list:
         """
         return itertools.chain(*self.grid)
 
-    def coord_iter(self) -> Iterator[Tuple[GridContent, int, int]]:
+    def coord_iter(self) -> Iterator[Tuple[Content, int, int]]:
         """ An iterator that returns coordinates as well as cell contents. """
         for row in range(self.width):
             for col in range(self.height):
                 yield self.grid[row][col], row, col  # agent, x, y
 
-    def neighbor_iter(
-        self, pos: Coordinate, moore: bool = True
-    ) -> Iterator[GridContent]:
+    def neighbor_iter(self, pos: Coordinate, moore: bool = True) -> Iterator[Agent]:
```

Comment:
Did I make a mistake when I annotated the return type? Seems like this change can be put in a separate commit for clarity.

Reply:
Yes, the return type annotations were wrong. I also just cached this. But despite its name `iter_cell_list_contents` does not return `Iterator[Content]`, but rather `Iterator[Agent]` (since empty cells are not iterated). And this trickles down to all neighbor functions. I'll see what I can do about separating the commits.

Reply:
Good idea to separate the commit. Found several more places were the return type was wrong.